README.md

```diff
@@ -4,132 +4,949 @@
+- **CLI Tool**: Comprehensive command-line interface (`kagent`) for all operations
+- **Kubernetes API**: Native Kubernetes API integration for GitOps workflows
+- **Multiple Frameworks**: Support for ADK, CrewAI, and LangGraph agent frameworks
+- **Local Development**: Full support for local development and testing
 
 ---
 
-## Getting started
+## 🎯 Why kagent?
 
-- [Quick Start](https://kagent.dev/docs/kagent/getting-started/quickstart)
-- [Installation guide](https://kagent.dev/docs/kagent/introduction/installation)
+### The Problem
 
-## Technical Details
+Organizations are increasingly looking to leverage AI agents for automation, but face significant challenges:
 
-### Core Concepts
+- **Infrastructure Complexity**: Deploying and managing AI agents requires complex infrastructure setup
+- **Security Concerns**: Ensuring secure access to Kubernetes resources and external APIs
+- **Operational Overhead**: Lack of standardized patterns for agent lifecycle management
+- **Integration Challenges**: Difficulties integrating agents with existing Kubernetes tooling
+- **Scalability Issues**: Challenges scaling agent deployments across multiple clusters
 
-- **Agents**: Agents are the main building block of kagent. They are a system prompt, a set of tools and agents, and an LLM configuration represented with a Kubernetes custom resource called "Agent". 
-- **LLM Providers**: Kagent supports multiple LLM providers, including [OpenAI](https://kagent.dev/docs/kagent/supported-providers/openai), [Azure OpenAI](https://kagent.dev/docs/kagent/supported-providers/azure-openai), [Anthropic](https://kagent.dev/docs/kagent/supported-providers/anthropic), [Google Vertex AI](https://kagent.dev/docs/kagent/supported-providers/google-vertexai), [Ollama](https://kagent.dev/docs/kagent/supported-providers/ollama) and any other [custom providers and models](https://kagent.dev/docs/kagent/supported-providers/custom-models) accessible via AI gateways. Providers are represented by the ModelConfig resource.
-- **MCP Tools**: Agents can connect to any MCP server that provides tools. Kagent comes with an MCP server with tools for Kubernetes, Istio, Helm, Argo, Prometheus, Grafana,  Cilium, and others. All tools are Kubernetes custom resources (ToolServers) and can be used by multiple agents.
-- **Observability**: Kagent supports [OpenTelemetry tracing](https://kagent.dev/docs/kagent/getting-started/tracing), which allows you to monitor what's happening with your agents and tools.
+### The Solution
 
-### Core Principles
+kagent addresses these challenges by providing a **Kubernetes-native framework** that makes deploying and managing AI agents as simple as deploying any other Kubernetes workload:
+
+#### 🚀 Quick Declarative Deployment
+
+Deploy agents in seconds using familiar Kubernetes tools and YAML manifests. No complex infrastructure setup required—just define your agent declaratively and apply it:
+
+```yaml
```

Comment:
this is not useful in at this location -- we're telling and showing users how Agent resource looks like, yet we haven't told them even how to install anything.